### ISSUE TITLE
Inappropriate praise message upon completing skills page

### ISSUE THREAD

**Issue by allydavies** [ucam-cl-dtg/isaac-app#457]
When a user achieves the mastery mark on a SKILLS BOOK page, an inappropriate praise message appears (and the user is easily directed away from the questions s/he was doing), but when you COMPLETE a page (i.e. get 100%), no praise message appears.
What's more this message keeps appearing every time an answer is entered (no matter whether the answer is right or wrong) once the mastery mark has been matched. This makes it very annoying trying to complete the board, as the message pops up, and bumps you back up the page, you have to close the message and scroll back to where you were.

Suggestion:
On achieving mastery mark, praise message could read:
"You've achieved the mastery mark on your board"
Green button message could read "Back to questions", and take them back to the questions (currently you CAN get straight back to the questions by clicking the closeX of the congratulations box, but if you click on the Back to Board button, you see a "Well done!" message over the hexagon for that page). If the closeX has to stay then give it the same function - back to the questions. In essence, don't give the user the "back to BOARD" option. They'll always have that button when they get to the bottom of the page / list of questions.
Make this message appear ONLY the first time the mastery mark is achieved.

On achieving 100%, praise message could read:
"Well done you've completed your board"
Maybe the "Congratulations!" banner could read "Congratulations - 100%!".

screenshots (not sure if these are working) and notes in case they're helpful:

![image](https://cloud.githubusercontent.com/assets/10402385/8874233/e88c6b5a-3206-11e5-98b8-7fdb4a11d156.png)
Message says "completed", but I have the mastery mark, not 100%.
Every subsequent answer entered ALSO gives me the same message (even if I get the answer to that question wrong), AND it bumps me up the page, so I have to scroll back to the question I was trying to answer.

If I click back to board, then this appears:
![image](https://cloud.githubusercontent.com/assets/10402385/8874243/13979fea-3207-11e5-9790-f31fc90c998a.png)

When I DO get 100%, I get nothing more than this:
![image](https://cloud.githubusercontent.com/assets/10402385/8874254/28bbce28-3207-11e5-82a9-f29d806a5e53.png)

**Comment by jsharkey13** [comment]
I think this could do with being addressed fairly soon - students will start to finish these boards, and this could cause quite a lot of confusion.

The book is currently by far the most used area of the site, judging by Google Analytics, and so odd behaviour when they use it isn't what we want.

**Comment by alistairstead3408** [comment]
This issue is a duplicate of #431. However, the current praise messages are:
- You've reached the passmark for your board
- You've completed your board

Would you like these changed to the following:
- You've achieved the mastery mark on your board
- Well done you've completed your board

**Comment by allydavies** [comment]
Further to your discussion with Lisa and James, yes I've cleared cache and the Mastery message appears only once. Thank you.

Have you agreed a wording with Lisa (she and I talked about "You've achieved the Skill Master mark on your board", to mirror the usage of Skill Master on the book pages?
For the 'completed' message, I suggested something mentioning '100%' or 'full marks', e.g. "Well done! You've achieved 100% on your board."

GREAT to see that the original fix works. Thank you.

**Comment by alistairstead3408** [comment]
Here's the commit for the updated messages:
https://github.com/ucam-cl-dtg/isaac-app/commit/c259c6b2be16aef64314b55fa80ce57f860b45bb

Currently for completing the board we have:
"You've completed your board"

I'm going to change this to
"You've achieved 100% on your board"

**Comment by allydavies** [comment]
I thought all aspects of the frustrating behaviour had been fixeed, but I think one remains or has resurfaced:

On achieving Skill Master mark, you get a message in a pop-up. There are TWO aspects that are still slightly irritating users (I know as some told me on Tuesday):

1. (seen as most frustrating) this message reappears after every correct answer entered once skill master is achieved. Would be much better to give skill master praise ONCE ONLY and then give 100% message only when 100% achieved.

2. (seen as less frustrating) If you close the pop-up with the X you can get straight on with entering the answers; BUT if you click the PROMINENT and ENTICING 'back to board' button, you have to click again on the hexagon for that book page, and then scroll (probably a long way) to your chosen question (whose accordion is probably closed again).

Was it the case that the skill master message USED to appear once only? I think it was and this has crept back in.

**Comment by allydavies** [comment]
Some screen shots to illistrate:

![image](https://cloud.githubusercontent.com/assets/10402385/13698804/4ad2a9b0-e76e-11e5-9e15-6ba9969153fd.png)

![image](https://cloud.githubusercontent.com/assets/10402385/13698806/51ea4ef6-e76e-11e5-902a-4d1d91afcf12.png)

![image](https://cloud.githubusercontent.com/assets/10402385/13698811/573d561e-e76e-11e5-98b8-813c8c9661b2.png)

**Comment by allydavies** [comment]
Another edge case

When a mastery task e.g. A1 is assigned within a board of other questions e.g. http://isaacphysics.org/s/2J1gqe
a student who achieves mastery on the A1 questions gets a message that (I think) implies they've completed it, not just mastered it AND which bumps them back to the board, rather than going through the remaining questions.
This will take some time to recreate (and I'm in a hurry to get to an event right now, sorry)

**Comment by jsharkey13** [comment]
I think we fixed this a while back, possibly by removing most of the pop-up causes. Reopen if it becomes and issue again!
